### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by iojcde**
Some performance improvements:
* Don't use Google fonts
* Convert images to WebP

Png images generally have very bad performance especially on the web, and converting all images to a modern format like WebP will most likely give a big performance boost. 

I tried to convert every image to webp, please tell me in case I missed a few.

Before           | After this PR
:-------------------------:|:-------------------------:
image | image

There's almost a 25% performance improvement, according to Google Pagespeed insights on mobile devices.

### CODE REVIEW

**Comment by alyx** [comment]
+1 to self-hosting the font. No good reason to use fonts.google.com anymore tbh -- serving locally provides us with fewer extra DNS queries, and reduces the dependencies on third-party platforms.

Regarding WebP, imo this change seems reasonable, but I think we may need to look at options for serving PNG to browsers which don't support WebP. It looks like Safari/macOS does not support WebP pre-Big Sur, which is still somewhat newer, and my understanding is that WebP support on QTWebKit / webkit2gtk can be questionable, based on per-distro configuration.

**Comment by JcdeA** [comment]
Thanks!

For the webp files, something like [this](https://stackoverflow.com/a/58888772) should work fine. I'll work on making the images show on legacy browsers

**Comment by knightss27** [comment]
Feel free to just make a new Svelte component that works with both the webp and png versions in that case, as it is probably the most concise solution. I tried a while back to integrate this with [svelte-image](https://github.com/matyunya/svelte-image) but their component messed up a lot of styling, so you could always try that. Also can I ask what you used for webp conversion? It's great to have all the existing files done, but in the future I should probably have the site create webp versions of the images on build, so converting isn't the job of whoever is adding the photo to the site. What you've done so far looks great though :)

**Comment by JcdeA** [comment]
> Feel free to just make a new Svelte component that works with both the webp and png versions in that case, as it is probably the most concise solution. I tried a while back to integrate this with [svelte-image](https://github.com/matyunya/svelte-image) but their component messed up a lot of styling, so you could always try that. Also can I ask what you used for webp conversion? It's great to have all the existing files done, but in the future I should probably have the site create webp versions of the images on build, so converting isn't the job of whoever is adding the photo to the site. What you've done so far looks great though :)

I used just plain cwebp on linux with GNU parallel to convert all png images to webp. Then I just deleted all the converted png files , and used sed to replace all .png urls to .webp. Not very complicated, even the GNU parallel script was copied from Stack Overflow.

**Comment by mplscorwin** [comment]
With both the font change and the change to the preferred image delivery format, I wonder if these would be things we can factor out at "build" time (e.g. when deploying the site).  It may be easier for some of our contributors that generally publish just content/pages (vs functionality/UI).  It's easier for those folks if they can upload in PNG (or other formats) and count on the site to actually sever the ideal mime-type (having been statically rendered at build time).

On the front side, taking the latest version of webfots available as of build-time could be preferable to creating a static instance we then have to check against it's upstream periodically.


Not preparing any patch of my own here; just some ideas :)  Thanks so much for interest and contributions (and for your work on FOSS projects, in general)!

**Comment by JcdeA** [comment]
I could probably make a script that converts images to webp that runs on build time

**Comment by JcdeA** [comment]
> Thanks so much for interest and contributions (and for your work on FOSS projects, in general)!

No, Thank YOU, and the Fosshost team for your work! :)

I hope Fosshost  can continue to support open source projects, with better infrastructure.

File: src/ambient.d.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ declare module "*.jpeg" {
 	export = value;
 }
 
-declare module "*.webp" {
+declare module "*.pdf" {
```
Comment (GrahamSH-LLK) [inline src/ambient.d.ts:26]: Did you intentionally write pdf?

Comment (JcdeA) [reply src/ambient.d.ts:26]: Nope, I have no idea why it became pdf

**Pull request closed without merging**